Data Structures/Queues/RingQueue.java

```diff
@@ -0,0 +1,94 @@
+public interface Queue {
+
+	public void add(Object object);
+
+	public Object first();
+
+	public Object remove();
+
+	public int size();
+
+}
+public class RingQueue implements Queue {
+	private Object[] objArray = null;
+	private int size;
+	private int front;
+	private int rear;
+
+	public RingQueue(int n) {
+		// Create an array of size n
+		objArray = new Object[n];
+		size = 0;
+		front = 0;
+		rear = 0;
+	}
+
+	@Override
+	public void add(Object obj) {
+		// type obj if there is empty space in the queue
+		// if there is no empty space in the queue throw new IllegalStateException ("The queue is full");
+		if (size == objArray.length)
```

Comment:
Please use curly braces. See here for if-else: http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html#449